src/incubator/WheelPicker/index.tsx

```diff
@@ -2,90 +2,164 @@
 // TODO: Support style customization
 // TODO: Support control of visible items
 import _ from 'lodash';
-import React, {useCallback, useRef} from 'react';
-import {TextStyle, FlatList, StyleProp} from 'react-native';
+import React, {useCallback, useRef, useState, useMemo} from 'react';
+import {TextStyle, FlatList, StyleProp, NativeSyntheticEvent, NativeScrollEvent} from 'react-native';
 import Animated from 'react-native-reanimated';
-import {onScrollEvent, useValues} from 'react-native-redash';
-
+import {useValues} from 'react-native-redash';
+import {Colors} from '../../../src/style';
 import View from '../../components/view';
+import Fader, {FaderPosition} from '../../components/fader';
 import {Constants} from '../../helpers';
 
 import Item, {ItemProps} from './Item';
 
 const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
 
 export interface WheelPickerProps {
+  /**
+   * Data source for WheelPicker
+   */
   items?: ItemProps[];
+  /**
+   * Describe the height of each item in the WheelPicker
+   */
   itemHeight?: number;
-  itemTextStyle?: StyleProp<TextStyle>;
-  onChange: (item: ItemProps, index: number) => void;
+  /**
+   * TextStyle for the focused row
+   */
+  activeItemTextStyle?: StyleProp<TextStyle>;
+  /**
+   * TextStyle for other, non-focused rows
+   */
+  inactiveItemTextStyle?: StyleProp<TextStyle>;
+  /**
+   * Event, on active row change
+   */
+  onChange: (index: number, item?: ItemProps) => void;
 }
 
-const WheelPicker = ({items, itemHeight = 48, itemTextStyle}: WheelPickerProps) => {
-  const height = itemHeight * 4;
-  const scrollview = useRef<Animated.ScrollView>();
+type WrappedItem = {
+  isActive: boolean;
+} & ItemProps;
+
+const WheelPicker = ({items, itemHeight = 48, activeItemTextStyle, inactiveItemTextStyle, onChange: onChangeEvent}: WheelPickerProps) => {
+  const height = itemHeight * 5;
+  const scrollView = useRef<Animated.ScrollView>();
   const [offset] = useValues([0], []);
-  const onScroll = onScrollEvent({y: offset});
```

Comment:
using onScrollEvent is actually very helpful for us.
It maps the scroll event on the native side and set whatever value we want. 
This is helpful cause it can save us from updating our state. 

In general the whole implementation of this component, using Reanimated, was planned to fix performance issue. 
Using Reanimate, or animated values in general is a great way to store values over the bridge and access them using interpolation to create various transitions and animation directly on the native thread. 

It's a little different approach from what we usually do